moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1822,12 +1807,32 @@ bool PlanningScene::processCollisionObjectMove(const moveit_msgs::CollisionObjec
     }
 
     collision_detection::World::ObjectConstPtr obj = world_->getObject(object.id);
+
+    // Update subframe poses
+    moveit::core::FixedTransformsMap subframe_poses = obj->subframe_poses_;
```

Comment:
This whole code snipped on subframe updates is only relevant if a shape update is actually performed below.
Hence, I think this code should be moved into the if condition below and a warning should be issued if a user just tries to move subframes (but no shapes).

Reply:
The data structures actually allow to handle the additional use-case to specify *only* subframe poses and no shapes.
It's not really an important use-case (maybe for tcp online estimation?), but I think we could include it here?
Either way, the current patch need updating as it mixes both interpretations, doing the computations independent and then update them only if new shape poses exist...

Reply:
If you just want to update subframes, you better use APPEND mode.

Reply:
Yes, the MOVE operation shouldn't add anything to the object. I removed the whole if block and just updated the world object.